oeps/oep-38-Data-Modeling.rst

```diff
@@ -0,0 +1,438 @@
+
+Definitions
+
+===========
+
+BI - Business intelligence. Technologies, applications and practices for
+the collection, integration, analysis, and presentation of business
+information. The purpose of Business Intelligence is to support better
+business decision making.
+
+Composite Key - in the context of relational databases, a composite key
+is a combination of two or more columns in a table that can be used to
+uniquely identify each row in the table. Uniqueness is only guaranteed
+when the columns are combined; when taken individually, the columns do
+not guarantee uniqueness.
+
+CUD - Create, Update, Delete. These are some of the actions that may
+happen to data in a database system.
+
+Data Best Practices - These practices are designed to help teams create
+rich and efficient data models within the edX ecosystem. They are not
+standards but guidelines to help teams think about how to store data.
+
+Data Dimension - A Data Dimension is a set of data attributes pertaining
+to something of interest to a business. Dimensions are things like
+"customers", "products", "stores" and "time". For users of Data
+Warehouses, data dimensions are entry points to numeric facts (e.g.
+sale, profit, revenue) that a business wishes to monitor.
+
+Data Modeling - A process used to define and analyze data requirements
+needed to support the business processes within the scope of
+corresponding information systems in organizations.
+
+Data Modeling Standards - The most basic and standard design principles
+for data modeling. These standards must be adhered to when creating new
+models or updating existing models in the edX ecosystem.
+
+IDA - “Independently Deployed Application”. Similar to a microservice or
+stand alone application.
+
+OpenedX Ecosystem
+
+------------------
+
+Used in this document, the phrase "OpenedX Ecosystem" includes the
+services that comprise and support the running of an OpenedX
+installation (e.g.,. edx.org). This includes:
+
+- The LMS and Studio
+
+- IDAs such as Notes, Ecommerce, and Forums
+
+- Backend systems such as data analytics and operational logging
+
+- Third party services used to extend functionality of applications,
+such as Optimizely, Google Analytics, and Sailthru
+
+Specification
+
+=============
+
+Guiding Principles
+------------------
+
+Data is an asset
+~~~~~~~~~~~~~~~~
+
+edX uses data to decide which people should receive marketing emails,
+who passes or fails a course, or how much to pay our partners. Our
+partners use our data to target users who fit certain criteria and to
+refine and improve their courses. The data we collect today may be used
+to gain new educational insights we have yet to realize. Decision making
+at edX should be data-driven and based on this collected data.
+
+This data is one of our most valuable assets and it should be and it
+should be a first-order concern. Save everything (disk is cheap). Fire
+lots of events. The more data we are able to collect about users,
```

Comment:
> Fire lots of events.

Is there a such thing as over-eventing? For example, if a single API view fires three redundant events, could that actually hinder data analysis?

Reply:
@kdmccormick I would say that there is so a thing as over-eventing, but that it is a better problem to have than not having the necessary data to make a decision. People should make reasonable attempts to not duplicate things, but if it's unclear that an event would be a duplicate I think we would rather lean towards having it.

Reply:
Removed the line about firing events.